components/Village/Edit/index.vue

```diff
@@ -0,0 +1,429 @@
+        <BaseInputText
+          v-model="form.name"
+          class="form-edit-village__form-group-field--label"
+          name="nama"
+          label="Nama Desa"
+          type="text"
+          placeholder="Masukkan nama desa"
+          autocomplete="false"
+          :error="!!(errors.villageName)"
+        />
+        <div v-if="errors.villageName" class="text-red-700">
+          {{ errors.villageName }}
+        </div>
+      </div>
+      <div class="form-edit-village__form-group">
+        <jds-select
+          v-model="form.city_id"
+          class="form-edit-village__form-group-field--label"
+          name="kabupaten/kota"
+          filterable
+          label="Kabupaten/Kota"
+          options-header="Kabupaten/Kota"
+          :options="optionsCity"
+          :error-message="errors.city"
+          placeholder="Pilih Kabupaten/Kota"
+        />
+      </div>
+      <div class="form-edit-village__form-group">
+        <jds-select
+          v-model="form.district_id"
+          :disabled="isDisabledOptionDistricts"
+          class="form-edit-village__form-group-field--label"
+          name="Kecamatan"
+          filterable
+          label="Kecamatan"
+          options-header="Kecamatan"
+          :options="optionsDistrict"
+          :error-message="errors.district"
+          placeholder="Pilih Kecamatan"
+        />
+      </div>
+      <div class="form-edit-village__form-group">
+        <jds-select
+          v-model="form.level"
+          class="form-edit-village__form-group-field--label"
+          name="Level Desa"
+          label="Level Desa"
+          :options="optionsLevel"
+          placeholder="Pilih Level"
+          :error-message="errors.villageLevel"
+        />
+      </div>
+      <div class="form-edit-village__form-group form-edit-village__form-group--position">
+        <div class="w-[250px]">
+          <BaseInputText
+            v-model="form.longitude"
+            class="form-edit-village__form-group-field--label"
+            name="longitude"
+            label="Longitude"
+            type="number"
+            placeholder="Contoh: 6.219212"
+            autocomplete="false"
+            :error="!!(errors.longitude)"
+          />
+          <div v-if="errors.longitude" class="text-red-700">
+            {{ errors.longitude }}
+          </div>
+        </div>
+        <div class="w-[250px]">
+          <BaseInputText
+            v-model="form.latitude"
+            class="form-edit-village__form-group-field--label"
+            name="latitude"
+            label="Latitude"
+            type="number"
+            placeholder="Contoh: 6.219212"
+            autocomplete="false"
+            :error="!!(errors.latitude)"
+          />
+          <div v-if="errors.latitude" class="text-red-700">
+            {{ errors.latitude }}
+          </div>
+        </div>
+      </div>
+    </form>
+  </BaseModal>
+</template>
+
+<script>
+import { isEqual, cloneDeep } from 'lodash'
```

Comment:
minor/discussion:

Hello, @doohanas I just want to discuss name import when using `lodash`. As far as I understand when using name import when importing a utility/function from `lodash` it's actually importing all the utility/function from `lodash` even if we're not using it. This is called [dead code](https://webpack.js.org/guides/tree-shaking/#add-a-utility)

IMO the easiest and safest method is to install and import  `lodash` utility that we actually need, ex:

```bash
npm i lodash.isequal
npm i lodash.clonedeep
``` 

and use them like this:
```js
import isEqual from 'lodash.isequal'
import cloneDeep from 'lodash.clonedeep'
```

I recommend we read about this topic from Adrian's research log, its full of really good insight and advice about how to improve our web app 👍🏼 

https://wiki.digitalservice.id/doc/3rd-use-tree-shaking-on-fortawesome-npm-package-pXifVamz6Z

Reply:
Hello @Ibwedagama,

Nice discussion and information⭐. I just know about the dead code. And I still wonder how we can detect the dead code in our app. And, how we can know about the package is support the tree-shaking. 

Based on your opinion, If we want to use the npm package and we don't want to install all the utilities, we have to know what utility we need, right?

Reply:
hello @doohanas sorry for the late reply, honestly I'm not sure how to effectively detect dead code but if we're using `webpack` we can analyze the build and see how extensive one library is being delivered to the user.

to do so we just need to install [webpack-bundle-analyzer](https://www.npmjs.com/package/webpack-bundle-analyzer)

![93f72404-b338-11e6-92d4-9a365550a701](https://user-images.githubusercontent.com/33661143/186050899-105f9fbb-f77a-4cbd-b5ba-ca5ae8d000ca.gif)


with this plugin, we can see how much size of the package/library, and then we should ask ourselves "Are we using all of the package/library functionality, and how we reduce the size of this package/library using tree-shaking?"

Note:

if we're using NuxtJS or Vue.js we can use these build commands:

**Vue.js**

```js
// Vue.js ./package.json

"build": "vue-cli-service build --report"
```

ref: https://techformist.com/bundle-size-vue-cli/

or using **NuxtJs**

```js
// nuxt.config.js

export default {
  build: {
    analyze: true,
    // or
    analyze: {
      analyzerMode: 'static'
    }
  }
}
```
ref: https://nuxtjs.org/docs/configuration-glossary/configuration-build/

Reply:
Thanks for the information about how to analyze the build and see how extensive one library is being delivered to the user.
So, it can be improved in the future for this project with @yoslie.